include/drivers/gpio.h

```diff
@@ -684,6 +691,17 @@ static inline int gpio_pin_configure(struct device *port, u32_t pin,
 		 || (flags & GPIO_OUTPUT) != 0,
 		 "Output needs to be enabled to be initialized low or high");
 
+	__ASSERT((flags & (GPIO_OUTPUT_INIT_LOW | GPIO_OUTPUT_INIT_HIGH))
+		 != (GPIO_OUTPUT_INIT_LOW | GPIO_OUTPUT_INIT_HIGH),
+		 "Output cannot be initialized low and high");
+
+	if (((flags & GPIO_OUTPUT_INIT_LOGICAL) != 0)
+	    && ((flags & (GPIO_OUTPUT_INIT_LOW | GPIO_OUTPUT_INIT_HIGH)) != 0)
+	    && ((flags & GPIO_ACTIVE_LOW) != 0)) {
+		flags ^= (GPIO_OUTPUT_INIT_LOW | GPIO_OUTPUT_INIT_HIGH);
```

Comment:
To save one operation we could write it as:
```
flags ^= (GPIO_OUTPUT_INIT_LOGICAL | GPIO_OUTPUT_INIT_LOW | GPIO_OUTPUT_INIT_HIGH);
```

Reply:
That's a little opaque for my taste, but OK.